### PR TITLE
Ref #23919 -- Replaced some os.path usages with pathlib.Path.

### PR DESCRIPTION
**Pull request by orf**
This continues cleaning up some `os.path` usages. I believe none of these changes break any public API's.

[Ticket](https://code.djangoproject.com/ticket/23919)

### CODE REVIEW

**Comment by srinivasreddy** [comment]
Remove os.walk too ?

**Comment by orf** [comment]
Where? I replaced listdir, replacing walk should be easy as well.

**Comment by srinivasreddy** [comment]
you can search this repo.

**Comment by orf** [comment]
Oh right, this isn't a comprehensive removal of all instances, just a
subset. I'll clean up any os.walk instances when I see them.

On 23 Jul 2017 10:27 am, "Srinivas Reddy Thatiparthy" <
notifications@github.com> wrote:

you can search this repo.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/django/django/pull/8797#issuecomment-317240466>, or mute
the thread

File: django/conf/__init__.py [hunk 1]
```diff
@@ -130,9 +131,8 @@ def __init__(self, settings_module):
         if hasattr(time, 'tzset') and self.TIME_ZONE:
             # When we can, attempt to validate the timezone. If we can't find
             # this file, no check happens and it's harmless.
-            zoneinfo_root = '/usr/share/zoneinfo'
-            if (os.path.exists(zoneinfo_root) and not
-                    os.path.exists(os.path.join(zoneinfo_root, *(self.TIME_ZONE.split('/'))))):
+            zoneinfo_root = Path('/usr/share/zoneinfo')
+            if (zoneinfo_root.exists() and not zoneinfo_root.joinpath(*self.TIME_ZONE.split('/'))):
```
Comment (vstoykov) [inline django/conf/__init__.py]: To match the old code this should be:
```
if (zoneinfo_root.exists() and not zoneinfo_root.joinpath(*self.TIME_ZONE.split('/')).exists()):
```
You are missed the second `.exists()`.

Comment (orf) [inline django/conf/__init__.py]: Thanks! good catch

**Comment by timgraham** [comment]
Tom, are you finished with the cleanup?

**Comment by orf** [comment]
I am finished with this section, there are more modules to refactor later though. I think pathlib makes most of the statements more readable, there are a few where it seems a bit worse off. Let me know what you think :+1:

I also figured doing them in more digestible pieces might be preferable than one big chunk, not sure if this is the preferred way for tickets like this.

**Pull request merged**